.circleci/is-version-number-acceptable.sh

```diff
@@ -1,26 +1,32 @@
 #! /usr/bin/env bash
 
-VERSION_CHANGE_TRIGGERS="setup.py MANIFEST.in openfisca_france"
-
-current_version=`python setup.py --version`
+if [[ $CIRCLE_BRANCH == master ]]
+then
+    echo "No need for a version check on master."
+    exit 0
+fi
 
-if git diff-index --quiet origin/master -- $VERSION_CHANGE_TRIGGERS ":(exclude)*.md"
-then exit 0  # there are no functional changes at all, the version is correct
+if $(dirname "$BASH_SOURCE")/detect-functional-changes.sh
+then
+    echo "No need for a version update."
+    exit 0
 fi
 
+current_version=`python setup.py --version`
+
 if git rev-parse --verify --quiet $current_version
 then
-    echo "Version $current_version already exists:"
+    echo "Version $current_version already exists in commit:"
     git --no-pager log -1 $current_version
     echo
     echo "Update the version number in setup.py before merging this branch into master."
     echo "Look at the CONTRIBUTING.md file to learn how the version number should be updated."
     exit 1
 fi
 
-if git diff-index --quiet origin/master CHANGELOG.md
+if ! $(dirname "$BASH_SOURCE")/detect-functional-changes.sh | grep --quiet CHANGELOG.md
```

Comment:
It is strange to me that `! detect-functional-changes.sh` means that changed were actually detected, as I read `!`as "not"

Reply:
Is the `grep --quiet CHANGELOG.md` applied in the `git diff-index --name-only` in `detect-functional-changes.sh`?

Reply:
The command on the left of `|` returns the list of files that have changed.
The grep checks if the Changelog is among them.
The `!` is here to because we want to raise an error if the changelog has _not_ been updated.

Reply:
So this line doesn't really care about the exit code of `detect-functional-changes.sh`. It's the `grep` exit code that is negated with `!`.

But I agree with you that [that line](https://github.com/openfisca/openfisca-france/blob/7309dae7ff31de462cd3ba58f17118efc3f38c84/.circleci/config.yml#L61) does not read well. Fixing that for all repos.